modules/unsupported/elasticsearch/src/main/java/org/geotools/data/elasticsearch/date/ElasticsearchDateConverter.java

```diff
@@ -425,8 +425,13 @@ public <T extends TemporalAccessor> T parse(String input, Class<T> type) {
                                     accessor.query(getTemporalQuery(LocalDateTime.class));
                             return (T) localDateTime.toInstant(ZoneOffset.UTC);
                         } else {
+                            int dayOfMonth = 1;
+                            if (accessor.isSupported(ChronoField.DAY_OF_MONTH)) {
```

Comment:
We just verified that the accessor supports the hour or day... but it may not support the day of month? :rofl: 
What kind of sorcery is this!